### PR TITLE
Non substantive ballot changes

### LINKED ISSUE #238: May 2018 Ballot Comment 32
## May 2018 Ballot Comment 32

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** Prefetch template
  **Type:** A-Q :question:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > MUST BE equivalent to the data the CDS Service would receive if it were making its own call to the EHR FHIR server using the parameterized prefetch template.
  
  **Comment:**
  How do you make a call using a prefetch template? We should first provide clear definitions of these concepts.
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive with Mod
  **Proposed Disposition Comment:**
  We will update the prefetch information (see comment #128) to make prefetch more clear and will address the comment here as well.
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #239: May 2018 Ballot Comment 33
## May 2018 Ballot Comment 33

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** Prefetch template
  **Type:** A-S :bulb:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > The CDS Service MUST NOT receive any prefetch template key that the EHR chooses not to satisfy;
  
  **Proposed Wording:**
  > The CDS Service Request SHALL not contain prefetch keys for which no data is provided.
  
  **Comment:**
  How can a service prevent receiving some content. This is a requirement on the sender, not the recipient.
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive with Mod
  **Proposed Disposition Comment:**
  Agree could be misread, perhaps restate the other direction as: The EHR MUST NOT send any... 
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #242: May 2018 Ballot Comment 36
## May 2018 Ballot Comment 36

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** Prefetch template
  **Type:** NEG :exclamation:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > data to populate a template prefetch key
  
  **Comment:**
  Please provide definition for prefetch key.
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive
  **Proposed Disposition Comment:**
  Agreed, will add definition for prefetch key and value
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #244: May 2018 Ballot Comment 38
## May 2018 Ballot Comment 38

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** Prefetch template
  **Type:** A-T :pencil2:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > To reduce the implementation burden on EHRs that support CDS Services, CDS Hooks recommends that prefetch queries only use a subset of the full functionality available in the FHIR specification. Valid prefetch URLs should only contain
  
  **Proposed Wording:**
  > To reduce the implementation burden on EHRs that support CDS Services, CDS Hooks RECOMMENDSthat prefetch queries only use a subset of the full functionality available in the FHIR specification. Valid prefetch URLs SHOULD only contain
  
  **Comment:**
  Use mandatory text where applicable.
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive
  **Proposed Disposition Comment:**
  We will update per the comment
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #246: May 2018 Ballot Comment 40
## May 2018 Ballot Comment 40

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** 
  **Type:** A-Q :question:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > To reduce the implementation burden on EHRs that support CDS Services, CDS Hooks recommends that prefetch queries only use a subset of the full functionality available in the FHIR specification. Valid prefetch URLs should only contain
  
  **Comment:**
  Prefetch URL ? New term – is this the same a prefetch template?
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive with Mod
  **Proposed Disposition Comment:**
  We will update the documentation to use the term prefetch template rather than prefetch URL.
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #257: May 2018 Ballot Comment 51
## May 2018 Ballot Comment 51

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** CDS Service Response - Card Attributes
  **Type:** NEG :exclamation:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > Allows a service to suggest a set of changes in the context of the current activity (e.g. changing the dose of the medication currently being prescribed, for the medication-prescribe activity). If used, the user MUST BE allowed to choose no more than one suggestion.
  
  **Comment:**
  How to deal with different services that address the same resource or are dependent on the same resource(s) for the calculation?
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive with Mod
  **Proposed Disposition Comment:**
  How conflicting or overlapping decision support from CDS Services should be handled in the EHR is outside of the specification.

We will add clarifying documentation noting that the specification does not address these types of scenarios and that both EHRs and CDS Services need to consider this in their integrations.
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #262: May 2018 Ballot Comment 56
## May 2018 Ballot Comment 56

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** CDS Service Response - Suggestion - Action
  **Type:** A-S :bulb:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > In activities like medication-prescribe where only one "content" resource is ever relevant, this field may be omitted.
  
  **Comment:**
  Medication-prescribe and order-review are hooks and as such not known to this spec. Remove or use an example-hook
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive with Mod
  **Proposed Disposition Comment:**
  We will remove all references to specific hooks and instead reference imagined example hooks. We will use org.example.<hook-name> as the hook names.
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #264: May 2018 Ballot Comment 58
## May 2018 Ballot Comment 58

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** CDS Services
  **Section:** CDS Service Response - Suggestion - Action
  **Type:** A-Q :question:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > For update, this holds the updated resource to modify from the current activity
  
  **Comment:**
  Activity, hook, or hook instance?
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive with Mod
  **Proposed Disposition Comment:**
  We will update this to clarify that update actions modify the given FHIR resource, whether that applies to a FHIR resource provided as part of the hook context or any FHIR resource available from the given FHIR server. 

We will also remove references to "current activity" as is misleading and not accurate in all scenarios.

We will also ensure that all references to CDS Hooks fields are using the code notation (backticks in Markdown).
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #274: May 2018 Ballot Comment 68
## May 2018 Ballot Comment 68

  Submitted by @bvdh from Philips Healthcare

  **Chapter:** Hooks
  **Section:** Hooks Definition Format - FHIR resoruces in context
  **Type:** A-Q :question:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Existing Wording:**
  > Table below All FHIR resources in contextMUST be based …
  
  **Comment:**
  Is this table and example or a spec element?
  
  ## Triage Information

  **Triage Notes:**
  Reviewed with Bas

  **Proposed Disposition:** Persuasive with Mod
  **Proposed Disposition Comment:**
  We will update the documentation to make it clear this table is an example
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #281: May 2018 Ballot Comment 75
## May 2018 Ballot Comment 75

  Submitted by @cmoesel from MITRE

  **Chapter:** Providing FHIR Resources to a CDS Service
  **Section:** 
  **Type:** A-S :bulb:
  **In Person Requested?** No
  
  **Existing Wording:**
  > FHIR resources requested in the CDS Service desciption are passed as key-value pairs, with each key matching a key described in the CDS Service description, and each value being a FHIR resource.
  
  **Proposed Wording:**
  > FHIR resources requested in the CDS Service desciption are passed as key-value pairs, with each key matching a key described in the CDS Service description, and each value being a FHIR resource (in the case of searches, a "searchset" Bundle containing a set of FHIR resources).
  
  **Comment:**
  On first read, the expectation that a prefetch value is a single resource can be confusing until you realize that search results come back as a single bundle containing multiple resources. I suggest the spec notes this to avoid confusion. My proposed wording is an example -- modify as you wish.
  
  ## :de: Köln May 2018 Working Group Vote

  @kpshek moved the following disposition, seconded by @olbrich.

  **Disposition:** Persuasive with Mod
  **Disposition Comment:**
  Agreed. We will add something like "For example, in the case where the prefetch template is a search, the prefetch query will return a Bundle"

  :+1: For: 9
  :expressionless: Abstain: 0
  :-1: Against: 0

  :tada: The motion passed! :tada:
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #282: May 2018 Ballot Comment 76
## May 2018 Ballot Comment 76

  Submitted by @cmoesel from MITRE

  **Chapter:** Providing FHIR Resources to a CDS Service
  **Section:** Prefetch Template
  **Type:** A-Q :question:
  **In Person Requested?** No
  
  **Existing Wording:**
  > If the EHR has no data to populate a template prefetch key, the prefetch template key MUST have a value of null.
  
  **Comment:**
  What is the expected result if the prefetch is a search query that the EHR supports, but has 0 results? Does the existing wording mean the result should be null (rather than a bundle w/ 0 entries)? Please clarify in spec - as I *expected* a 0-result bundle would be appropriate to indicate the search succeeded but had 0 results..
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #287: May 2018 Ballot Comment 81
## May 2018 Ballot Comment 81

  Submitted by @cmoesel from MITRE

  **Chapter:** Hooks
  **Section:** Hook context and prefetch
  **Type:** A-Q :question:
  **In Person Requested?** No
  
  **Existing Wording:**
  > If a context field can be tokenized, the value of the context field MUST be a data type that can placed into a FHIR query (eg, string, number, etc).
  
  **Comment:**
  Are multi-cardinality (e.g., array) fields allowed as tokens? If so, are they interpreted as an "OR" search?
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #289: May 2018 Ballot Comment 83
## May 2018 Ballot Comment 83

  Submitted by @cmoesel from MITRE

  **Chapter:** Hooks
  **Section:** Hook version
  **Type:** NEG :exclamation:
  **In Person Requested?** No
  
  **Existing Wording:**
  > Change of prefetch token status of an existing context field - Patch
  
  **Proposed Wording:**
  > Change of prefetch token status of an existing context field - Major
  
  **Comment:**
  Changing prefetch token status from "Yes" to "No" would impact any existing services that already use the context field in a prefetch template -- as it would render the prefetch invalid.
 
 Changing prefetch token status from "No" to "Yes" would impact EHRs as they now need to support an additional token in prefetch templates that they didn't need to support before.
 
 These seem to be backwards-incompatible changes to me, and so should indicate a Major revision.
  
  ## :de: Köln May 2018 Working Group Vote

  @kpshek moved the following disposition, seconded by @brynrhodes.

  **Disposition:** Persuasive with mod
  **Disposition Comment:**
  We will fix this entry in the table to indicate prefetch token changes (boolean status) to required context fields are breaking changes (major). Prefetch token changes to optional context fields are not breaking changes.

Additionally, we will document that the intention behind the table is to outline possible breaking changes and when we are notified of additional scenarios, we will add them. 

  :+1: For: 19
  :expressionless: Abstain: 3
  :-1: Against: 0

  :tada: The motion passed! :tada:
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #298: May 2018 Ballot Comment 92
## May 2018 Ballot Comment 92

  Submitted by Kanwarpreet Sethi on behalf of KP Sethi from Lantana Consulting Group

  **Chapter:** Hook context and prefetch
  **Section:** 
  **Type:** A-S :bulb:
  **In Person Requested?** No
  
  **Comment:**
  Hook context and prefetch discussion is somewhat confusing, and the examples aren't very helpful since they repeat a lot of the same information.
 I would suggest using a simple running example and showing the context, the prefetch, including what their definitions look like so its easier to understand what they look like.
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #300: May 2018 Ballot Comment 94
## May 2018 Ballot Comment 94

  Submitted by Kanwarpreet Sethi on behalf of KP Sethi from Lantana Consulting Group

  **Chapter:** General
  **Section:** 
  **Type:** NEG :exclamation:
  **In Person Requested?** Yes :bust_in_silhouette:
  
  **Comment:**
  A section that defining "Context" should be included since it seems to be a major part of the hook/cds service. Currently context isnt defined until the HTTP request section.
  
  ## :de: Köln May 2018 Working Group Vote

  @kpshek moved the following disposition, seconded by @isaacvetter.

  **Disposition:** Persuasive with Mod
  **Disposition Comment:**
  We will move the hooks.md into 1.0.md page.

  :+1: For: 11
  :expressionless: Abstain: 0
  :-1: Against: 0

  :tada: The motion passed! :tada:
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #313: May 2018 Ballot Comment 107
## May 2018 Ballot Comment 107

  Submitted by @kensaku-kawamoto on behalf of @euvitudo from University of Utah

  **Chapter:** Prefetch Template
  **Section:** https://cds-hooks.org/specification/1.0/
  **Type:** A-S :bulb:
  **In Person Requested?** No
  
  **Existing Wording:**
  > The prefetch template is a dictionary
  
  **Proposed Wording:**
  > The prefetch template is a listing
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #314: May 2018 Ballot Comment 108
## May 2018 Ballot Comment 108

  Submitted by @kensaku-kawamoto on behalf of @euvitudo from University of Utah

  **Chapter:** Prefetch Template
  **Section:** https://cds-hooks.org/specification/1.0/
  **Type:** A-Q :question:
  **In Person Requested?** No
  
  **Existing Wording:**
  > The EHR may compute an efficient set of prefetch templates from multiple CDS Services, thereby reducing the number of network calls to a minimum
  
  **Comment:**
  What are the implications of this for a service wtihin this group? Will it receive more data than requested?
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #316: May 2018 Ballot Comment 110
## May 2018 Ballot Comment 110

  Submitted by @kensaku-kawamoto on behalf of @euvitudo from University of Utah

  **Chapter:** Prefetch Template
  **Section:** https://cds-hooks.org/specification/1.0/
  **Type:** A-Q :question:
  **In Person Requested?** No
  
  **Existing Wording:**
  > The resulting response, which MUST BE rendered in a single page -- no "next page" links allowed --
  
  **Comment:**
  Is this allowed in FHIR? If not, any reasons why it's specifically mentioned here?
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #343: May 2018 Ballot Comment 137
## May 2018 Ballot Comment 137

  Submitted by Nell Lapres on behalf of @isaacvetter from Epic

  **Chapter:** 
  **Section:** 
  **Type:** A-S :bulb:
  **In Person Requested?** No
  
  **Comment:**
  In the Example prefretch templates section, the prefetch template keys would be more readable if they were named something like "patient", "hemoglobin-a1c" and "user"; rather than "p", "a1c" and "u". While the documentation's goal to illustrate that the prefetch keys can be named whatever is laudable, it's actually not that important. In practice, it would be a benefit to the community if "patient", "user", etc were commonly used by CDS services to describe the current patient and user in prefretch templates.
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #344: May 2018 Ballot Comment 138
## May 2018 Ballot Comment 138

  Submitted by Walter Suarez from Kaiser Permanente

  **Chapter:** Specifications
  **Section:** Overview
  **Type:** A-T :pencil2:
  **In Person Requested?** No
  
  **Comment:**
  MUST BE vs MUST be - similarly, there are several other instances where "MUST BE" is used, instead of "MUST be"
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### LINKED ISSUE #349: May 2018 Ballot Comment 143
## May 2018 Ballot Comment 143

  Submitted by Walter Suarez from Kaiser Permanente

  **Chapter:** 
  **Section:** Prefetch Query - Prefetch Query Restrictions
  **Type:** A-S :bulb:
  **In Person Requested?** No
  
  **Comment:**
  "Valid pefetch URLs should only contain:…" should it not be "Valid prefetch URLs MUST only contain:"?
  
  ---
  _This issue was imported by @cds-hooks-bot from the [consolidated CDS Hooks May 2018 ballot spreadsheet](https://bit.ly/cds-hooks-may-2018-ballot-comments)._

### PR DESCRIPTION
**Pull request by brynrhodes**
Addresses the following ballot comment issues:

Fixes #238: Clarified how prefetch templates are processed
Fixes #239: Clarified prefetch data conformance language
Fixes #242: Added definition for prefetch key
Fixes #244: Updated conformance language usage in prefetch restrictions
Fixes #246: Clarified prefetch limitations are on prefetch templates
Fixes #257: Added clarification on responses and notes about multiple services from the same hook
Fixes #262: Added examples for each action type
Fixes #264: Clarified documentation for actions
Fixes #274: Clarified hook context table is an example
Fixes #281: Clarified prefetch response resources may be a searchset bundle
Fixes #282: Provided rationale for the use of __null__ in the prefetch response
Fixes #287: Clarified allowable data types for context fields
Fixes #289: Changed prefetch token status change to Major
Fixes #298: Added examples to hook/prefetch discussion
Fixes #300: Added an overview section discussing CDS Hooks components
Fixes #313: Clarified definitions of prefetch template, prefetch token, and prefetch key
Fixes #314: Clarified prefetch behavior for multiple services attached to the same hook
Fixes #316: Added rationale for not allowing next page links in prefetch data
Fixes #344: Clarified distinction between context and prefetch
Fixes #343: Updated prefetch example to use full key names
Fixes #349: Corrected capitalization of conformance language in prefetch description

### CODE REVIEW

File: docs/specification/1.0.md [hunk 1]
```diff
@@ -736,9 +736,9 @@ Field | Optionality | Prefetch Token | Type | Description
 
 #### FHIR resources in context
 
-When potentially multiple FHIR resources value a single context field, these resources SHOULD be formatted as a FHIR Bundle. For example, multiple FHIR resources are necessary to describe all of the orders under review in the `order-review` hook's `orders` field. Hook definitions SHOULD prefer the use of FHIR Bundles over other bespoke data structures.
+When a context field may contain multiple FHIR resources, these resources SHOULD be formatted as a FHIR Bundle. For example, multiple FHIR resources are necessary to describe all of the orders under review in the `order-review` hook's `orders` field. Hook definitions SHOULD prefer the use of FHIR Bundles over other bespoke data structures.
```
Comment (brettmarquard) [inline docs/specification/1.0.md]: a bit awkward -- When a context field contains multiple FHIR resources

File: docs/specification/1.0.md [hunk 2]
```diff
@@ -625,15 +625,15 @@ Note that each hook (e.g. `medication-prescribe`) represents something the user
...
 
 Additionally, consider a PGX CDS Service and a Zika screening CDS Service, each of which is subscribed to the same hook. The context data specified by their shared hook should contain data relevant to both CDS Services; however, each service will have other specific data needs that will necessitate disparate prefetch requests. For instance, the PGX CDS Service likely is interested in genomics data whereas the Zika screening CDS Service will want Observations.
 
-In summary, context is data specific to a hook and universally relevant to all CDS Services subscribed to said hook. Prefetch data is unique to individual CDS Services and supplements the data from context.
+In summary, context is defined by the hook and typically relevant to all CDS Services subscribed to said hook. Prefetch data is defined by each CDS Service to support the specific behavior provided by the service.
```
Comment (brettmarquard) [inline docs/specification/1.0.md]: Thanks - this is better, my brain still struggles a bit with these two concepts. the use of word 'typically relevant' throws me off a bit.  Isn't SHALL in a CDS hooks definition context required?

Is this correct: Context is standardized and published with the hook definition. Prefetch is specific to the vendor providing the CDS service and is not standardized.?

File: docs/specification/1.0.md [hunk 3]
```diff
@@ -199,13 +199,15 @@ The prefetch tokens defined by this `example-hook` would be `{{context.patientId
...
 - The EHR MAY have some of the desired prefetched data already in memory, thereby removing the need for any network call
-- The EHR MAY compute an efficient set of prefetch templates from multiple CDS Services, thereby reducing the number of network calls to a minimum
+- The EHR MAY compute an efficient set of prefetch templates from multiple CDS Services, thereby reducing the number of calls to a minimum
 - The EHR MAY satisfy some of the desired prefetched templates via some internal service or even its own FHIR server.
 
 The EHR SHALL deny access if the requested resource is outside the user's authorized scope.
 
 Regardless of how the EHR satisfies the prefetch templates (if at all), the prefetched data given to the CDS Service MUST be equivalent to the data the CDS Service would receive if it were making its own call to the EHR FHIR server using the parameterized prefetch template.
 
+> Note that this means that CDS services will receive only the information they have requested and are authorized to receive. Prefetch data for other services registered to the same hook MUST NOT be provided.
```
Comment (brettmarquard) [inline docs/specification/1.0.md]: Would you be willing to clarity 'Prefetch data for other services registered to the same hook MUST NOT be provided.'

Do you mean unauthorized?

File: docs/specification/1.0.md (hunk 1 above)
Comment (brynrhodes) [reply docs/specification/1.0.md]: How about "For context fields that may contain multiple FHIR resources, the field SHOULD be defined as a FHIR Bundle, rather than as an array of FHIR resources."?

File: docs/specification/1.0.md (hunk 2 above)
Comment (brynrhodes) [reply docs/specification/1.0.md]: How about "In summary, context is defined by the hook because it is dictated by the information available at the point in the workflow at which the hook is triggered. Prefetch data is defined by each CDS Service because it is specific to the information that service needs in order to process."

I'd say your statement is correct, except that I wouldn't say "standardized", I'd just say "defined by".

File: docs/specification/1.0.md (hunk 3 above)
Comment (brynrhodes) [reply docs/specification/1.0.md]: It's even stronger than unauthorized. A service may be authorized to view some piece of data, but if it did not ask for it in it's prefetch, it should not receive it. So how about if we add a sentence at the end of the note:

"In other words, services should receive ALL and ONLY the data they requested in their prefetch and for which they are authorized recipients."

**Comment by brynrhodes** [comment]
@brettmarquard , I just committed some additional changes to the prefetch templates section that I hope address your concerns. Please take a look and let me know.

File: docs/specification/1.0.md [hunk 4]
```diff
@@ -736,9 +736,9 @@ Field | Optionality | Prefetch Token | Type | Description
...
 
-Often, context is populated with in-progress or in-memory data that may not yet be available from the FHIR server. For example, `medication-prescribe` and `order-review` define context fields containing FHIR resources that represent draft resources. In this case,  the EHR should only provide these draft resources and not the full set of orders available from its FHIR server. The CDS service MAY pre-fetch or query for FHIR resources with other statuses. 
+Often, context is populated with in-progress or in-memory data that may not yet be available from the FHIR server. For example, `medication-prescribe` and `order-review` define context fields containing FHIR resources that represent draft resources. In this case, the EHR should only provide these draft resources and not the full set of orders available from its FHIR server. The CDS service MAY pre-fetch or query for FHIR resources with other statuses.
```
Comment (kpshek) [inline docs/specification/1.0.md]: Reading this caused me to log #436

File: docs/specification/1.0.md (hunk 2 above)
Comment (brettmarquard) [reply docs/specification/1.0.md]: Can we say 'In summary, context is specified in the hook definition to guide developers on the information available at the point in the workflow when the hook is triggered. Prefetch data is defined by each CDS Service because it is specific to the information that service needs in order to process

File: docs/specification/1.0.md (hunk 3 above)
Comment (brettmarquard) [reply docs/specification/1.0.md]: In other words, services SHALL only receive the data they requested in their prefetch and for which they are authorized recipients.

File: docs/specification/1.0.md [hunk 5]
```diff
@@ -193,30 +193,29 @@ Similarly, each EHR will decide what FHIR resources to authorize and to prefetch
...
-The `prefetch` field of a CDS Service description defines the set of prefetch templates for that service, providing a _prefetch key_ for each one that is used to provide the prefetch data in the CDS request.
+```
+Observation?patient={{context.patientId}}&code=4548-4&_count=1&sort:desc=date
+```
 
-Prefetch tokens MUST be delimited by `{{` and `}}`, MUST be named based upon the field they correspond to, and MUST have a primitive value.
+To allow for prefetch templates that are dependent on the workflow context, prefetch templates may include references to context using _prefetch tokens_. In the above example, `{{context.patientId}}` is a prefetch token. A prefetch token is a placeholder in a prefetch template that is replaced by a value from the hook's context to construct the FHIR URL used to request the prefetch data. 
 
-Individual hooks specify which of their `context` fields can be used as prefetch tokens. Only root-level fields with a primitive value within the `context` object are eligible to be used as prefetch tokens.
+Prefetch tokens MUST be delimited by `{{` and `}}`, and MUST contain only the qualified path to a hook context field.
```
Comment (brettmarquard) [inline docs/specification/1.0.md]: Add link here to Prefetch tokens section

Comment (brynrhodes) [reply docs/specification/1.0.md]: This is already in the prefetch tokens section, is there another section you're thinking of that we should link to here?

File: docs/specification/1.0.md (hunk 4 above)
Comment (brynrhodes) [reply docs/specification/1.0.md]: Agreed, happy to apply that here.

File: docs/specification/1.0.md [hunk 6]
```diff
@@ -162,62 +183,70 @@ curl
...
 
 An EHR MAY choose to honor some or all of the desired prefetch templates, and is free to choose the most appropriate source for these data. For example:
 
 - The EHR MAY have some of the desired prefetched data already in memory, thereby removing the need for any network call
-- The EHR MAY compute an efficient set of prefetch templates from multiple CDS Services, thereby reducing the number of network calls to a minimum
+- The EHR MAY compute an efficient set of prefetch templates from multiple CDS Services, thereby reducing the number of calls to a minimum
 - The EHR MAY satisfy some of the desired prefetched templates via some internal service or even its own FHIR server.
 
-The EHR SHALL deny access if the requested resource is outside the user's authorized context.
+The EHR SHALL deny access to the request resource if it is outside the user's authorized scope.
```
Comment (kpshek) [inline docs/specification/1.0.md]:
```suggestion
The EHR SHALL deny access to the requested resource if it is outside the user's authorized scope.
```

**Comment by kpshek** [comment]
The changes in this PR also fixed #226
